node/v2/handler.js

```diff
@@ -325,9 +331,15 @@ TChannelV2Handler.prototype.handleCallResponse = function handleCallResponse(res
     if (handled) {
         if (res.state === States.Streaming) {
             self.streamingRes[res.id] = res;
+            res.finishEvent.on(onResponseDone);
+            res.errorEvent.on(onResponseDone);
         }
```

Comment:
if we do else branch, then we need to do it both on if (handled)  and if (res.state === States.Streaming).

And as @kriskowal pointed out, the "else" approach is less intuitive.